internal/mage/sdk/php.go

```diff
@@ -116,15 +116,15 @@ func (t PHP) Publish(ctx context.Context, tag string) error {
 		gitUserEmail = "hello@dagger.io"
 	}
 
-	_, err = util.GoBase(c).
+	git := util.GoBase(c).
 		WithExec([]string{"apk", "add", "-U", "--no-cache", "git"}).
 		WithExec([]string{"git", "config", "--global", "user.name", gitUserName}).
 		WithExec([]string{"git", "config", "--global", "user.email", gitUserEmail}).
-		WithSecretVariable("GITHUB_PAT", c.SetSecret("GITHUB_PAT", encodedPAT)).
-		WithExec([]string{
-			"sh", "-c",
-			`git config --global http.https://github.com/.extraheader "AUTHORIZATION: Basic $GITHUB_PAT"`,
-		}).
+		WithEnvVariable("GIT_CONFIG_COUNT", "1").
+		WithEnvVariable("GIT_CONFIG_KEY_0", "http.https://github.com/.extraheader").
+		WithSecretVariable("GIT_CONFIG_VALUE_0", c.SetSecret("GITHUB_HEADER", fmt.Sprintf("AUTHORIZATION: Basic %s", encodedPAT)))
+
+	_, err = git.
```

Comment:
Why break the chain if `git` isn't needed after this point?

Reply:
Avoided this for both php and go.